### PR TITLE
Update _message.py - issue #2606

### LINKED ISSUE #2606: [BUG] Message.photo is set to an empty list when not provided
### Steps to reproduce
1. Generate a Message object without photo set.

2. Check the photo parameter

### Expected behaviour
.photo should be None

### Actual behaviour
.photo is an empty list

Wrong line: https://github.com/python-telegram-bot/python-telegram-bot/blob/1fdaaac8094c9d76c34c8c8e8c9add16080e75e7/telegram/message.py#L515

### PR DESCRIPTION
**Pull request by nivramam**
closes #2606

### Checklist for PR
- [x] modified `message.photo = None` instead of an empty list `[]`
- [ ] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)

### CODE REVIEW

**Comment by Bibo-Joshi** [comment]
Hi. So for the change is okay and the tests run fine - I don't know why you had issues locally. So you can go ahead and make these kind of change everywhere where necessary :) Please also make sure to add versioning directives and we should also add some tests for the changed attributes to make sure that they're `None` and not `[]`. Please reinstate the pull request template for that which you deleted ;)

**Comment by Bibo-Joshi** [comment]
I'm not sure what you mean. You haven't opened a PR on this repo before.
What I meant is that there is a checklist in the text field when you create a new PR and we want you to keep it there. Please copy the relevant part from [here](https://raw.githubusercontent.com/python-telegram-bot/python-telegram-bot/master/.github/pull_request_template.md) into the PR description

**Comment by nivramam** [comment]
Oh fine you meant that. I'll edit the description right away..

**Comment by nivramam** [comment]
So here I've drafted to add the following highlighted text since the others aren't relevant here in my case. 
image
That's fine right?

P.S; Since this is the first ever time I am contributing please excuse me for the doubts that keep arising from me. Thanks :)

**Comment by nivramam** [comment]
Right then

**Comment by nivramam** [comment]
Hi, having a qq, while I was resolving a similar `[]` empty list issue in `replykeyboardmarkup`. Have described the same over here https://github.com/python-telegram-bot/python-telegram-bot/issues/2606#issuecomment-968329261

**Comment by nivramam** [comment]
Hi so I have added tests for the already implemented changes (test_message.py). Apart from that, the above mentioned addition of a null check on keyboardmarkup too I have committed to review the checks. Thx! And also I am in the process of reviewing other similar changes. Will update here as and when I complete them.

File: telegram/_replykeyboardmarkup.py [hunk 1]
```diff
@@ -114,8 +114,9 @@ def __init__(
         self.one_time_keyboard = bool(one_time_keyboard)
         self.selective = bool(selective)
         self.input_field_placeholder = input_field_placeholder
-
-        self._id_attrs = (self.keyboard,)
+        # ensure keyboard is notnull before assigning
+        if self.keyboard:
+            self._id_attrs = (self.keyboard,)
```
Comment (Bibo-Joshi) [inline telegram/_replykeyboardmarkup.py:119]: I don't think this check is necessary. `keyboard` is a mandatory argument and if the user passes something that's not a list of lists of buttons, that's their fault.
Additionally, the sanity check in line 95 alread checks the "list of lists" part. So if we get here, `self.keyboard` is `[[]]` in the worst case, which will pass the `if self.keyboard` check.

**Review by Bibo-Joshi — COMMENTED** [review]
Hi. Sorry for the late reply. I addressed your question on `ReplyKeyboardMarkup` below.
I'll do a more thorough review once you indicate that you've made all required changes :)

File: telegram/_message.py [hunk 2]
```diff
@@ -468,7 +468,7 @@ def __init__(
         self.audio = audio
         self.game = game
         self.document = document
-        self.photo = photo or []
+        self.photo = photo or None
```
[the remark this answers is not in the document]
Comment (nivramam) [reply telegram/_message.py:471]: Right yes, shall modify it :)

File: telegram/_replykeyboardmarkup.py (hunk 1 above)
Comment (nivramam) [reply telegram/_replykeyboardmarkup.py:119]: Right. Yeah the check at 95 was there, but I thought for ensuring the null check would be necessary. 

But thanks to your comment, I just reviewed and realized it doesn't get to null anywhere. Sure will do changes and let you know

Comment (nivramam) [reply telegram/_replykeyboardmarkup.py:119]: And regarding conflicts there is a huge number of conflicts showing up now but wasn't there when I pushed yesterday.. Is it for me alone?

Comment (harshil21) [reply telegram/_replykeyboardmarkup.py:119]: just pull from v14 into this branch, it should be fine :)

Comment (nivramam) [reply telegram/_replykeyboardmarkup.py:119]: Yes, I have tried to do it. I was prompted to create a PR and I did so, but now I am stuck here. Could you pls give me a solution so that I have the latest V14? Thanks in advance! 

P.S: Due to my access issue I was unable to login to Github and hence the delay. 

Here's the screen of the PR that I created when prompted.
image

**Comment by Bibo-Joshi** [comment]
Hi. We discussed this issue in the dev chat again and had a bit of a change of heart. While the main argument to change to `None` was to be uniform with the other attributes, the benfits of `[]` are compelling and the issue of uniformness can be largely circumvented by properly adapting the documentation.  E.g. the docstring for `Message.photo` currently reads
> photo (List[:class:`telegram.PhotoSize`]): Optional. Available sizes of the photo.

and should be changed to something like
> photo (List[:class:`telegram.PhotoSize`]): Available sizes of the photo. This list is empty if the message doesn't contain a photo.

@nivramam I'm sorry that we reverted our decision only after you started to work on this PR. I hope that you haven't put too much time into it and that you can understand our decision. If your are interested, you're very welcome to instead PR for the mentioned updates of the documentation. In this case, please fetch the current `doc-fixes` branch and create a new branch based on that to add your changes to.

**Comment by nivramam** [comment]
> Hi. We discussed this issue in the dev chat again and had a bit of a change of heart. While the main argument to change to `None` was to be uniform with the other attributes, the benfits of `[]` are compelling and the issue of uniformness can be largely circumvented by properly adapting the documentation. E.g. the docstring for `Message.photo` currently reads
> 
> > photo (List[:class:`telegram.PhotoSize`]): Optional. Available sizes of the photo.
> 
> and should be changed to something like
> 
> > photo (List[:class:`telegram.PhotoSize`]): Available sizes of the photo. This list is empty if the message doesn't contain a photo.
> 
> @nivramam I'm sorry that we reverted our decision only after you started to work on this PR. I hope that you haven't put too much time into it and that you can understand our decision. If your are interested, you're very welcome to instead PR for the mentioned updates of the documentation. In this case, please fetch the current `doc-fixes` branch and create a new branch based on that to add your changes to.

Oh fine. Yes no issues on my side since I didn't have access to GitHub in the last month so no more updates from my side. 

Sure will try to PR the mentioned updates. Thanks!
